### PR TITLE
Optimize env interface and separate responsibilities.

### PR DESCRIPTION
**Pull request by wenxinnnnn**
Refer to #200 .
CLI remains unchanged.
Restful api is split into two interfaces.

### CODE REVIEW

File: e2e/apiserver/apiserver_test.go [hunk 1]
```diff
@@ -21,35 +19,39 @@ import (
 )
 
 var (
-	envHelloMeta = types.EnvMeta{
-		Name:      "env-e2e-hello",
+	envHelloMeta = apis.Environment{
```
Comment (zzxwill) [inline e2e/apiserver/apiserver_test.go:22]: What's the purpose of this change?

File: e2e/commonContext.go [hunk 2]
```diff
@@ -215,7 +214,7 @@ var (
 				var r apis.Response
 				err = json.Unmarshal(result, &r)
 				gomega.Expect(http.StatusOK).Should(gomega.Equal(r.Code))
-				output := fmt.Sprintf("Create env succeed, current env is " + envMeta.Name + " namespace is " + envMeta.Namespace)
```
Comment (zzxwill) [inline e2e/commonContext.go:218]: I think the former expected output is more specific.

File: pkg/oam/env.go [hunk 3]
```diff
@@ -34,11 +35,18 @@ func GetEnvByName(name string) (*types.EnvMeta, error) {
 	return &meta, nil
 }
 
+//Create or update env.
+//If it does not exist, create it and switch to the new env.
+//If it exists, update it and switch to the new env.
 func CreateOrUpdateEnv(ctx context.Context, c client.Client, envName string, namespace string) (error, string) {
```
Comment (zzxwill) [inline pkg/oam/env.go:41]: - I'm confused with func names  between `CreateEnv` and `UpdateEnv`, and `CreateOrUpdateEnv`.
- API and Cli should depend the same function in oam package, but `CreateOrUpdateEnv` only used by API.

**Comment by xiaoyuaiheshui** [comment]
I'll fix the dashboard to fit this API change.

File: e2e/commonContext.go (hunk 2 above)
Comment (wenxinnnnn) [reply e2e/commonContext.go:216]: Now the create an env interface does not provide the ability to switch env.

File: e2e/apiserver/apiserver_test.go (hunk 1 above)
Comment (wenxinnnnn) [reply e2e/apiserver/apiserver_test.go:22]: Environment object provides input parameter verification of api interface

File: pkg/oam/env.go (hunk 3 above)
Comment (wenxinnnnn) [reply pkg/oam/env.go:41]: API and Cli use different user habits, so I suggest that cli still retains the ability to switch while creating

Comment (zzxwill) [reply pkg/oam/env.go:41]: @wonderflow Please also take a look at this design.

File: e2e/apiserver/apiserver_test.go (hunk 1 above)
Comment (zzxwill) [reply e2e/apiserver/apiserver_test.go:22]: types.EnvMeta could not?

File: e2e/commonContext.go (hunk 2 above)
Comment (zzxwill) [reply e2e/commonContext.go:216]: I got it. @wonderflow is it okay with your initial design?

File: e2e/apiserver/apiserver_test.go (hunk 1 above)
Comment (wenxinnnnn) [reply e2e/apiserver/apiserver_test.go:22]: types.EnvMeta It's domain entity. Environment is similar to dto,in the future, the interface can carry more expressive force

**Review by wonderflow — APPROVED** [review]
